### ISSUE TITLE
Provide API for dual evaluation

### ISSUE THREAD

**Issue by wence-** [FInAT/FInAT#21]
In many cases we'd like to be able to evaluate dual basis functionals in physical space.  For example, computing restrictions in multigrid, interpolations, etc...  We can't guarantee up front that we know where in the reference cell we'd like to evaluate the nodes, so we need code to do it.  This kind of stuff is currently dotted around in various places (e.g. evaluate_basis in ffc, point evaluation in Firedrake).  I think we need to be able to ask for gem to evaluate (maybe only in reference space, since finat doesn't own pullbacks right now?) nodes at a provided (runtime) array of points.

Thoughts? Clarifications?

**Comment by miklos1** [comment]
`evaluate_basis` is for point evaluations, dual evaluation is `evaluate_dofs`.

**Comment by dham** [comment]
Careful here. Nodes are functionals in V^* so they take functions in V, not points. Basis functions can be evaluated at points, but that's not the same thing.

I agree that we need the ability to evaluate nodes. However I think that node evaluation probably takes another finite element as an argument and returns the GEM for the evaluation of the nodes of  this element on the basis functions of  the other element.

**Comment by wence-** [comment]
I agree with that idea.  How does that compose with physical-to-reference transformations?  Probably I need to write some maths...
